timeline-chart/src/time-graph-font-controller.ts

```diff
@@ -0,0 +1,60 @@
+import * as PIXI from "pixi.js";
+
+export class FontController {
+    private fontFamily: string;
+    private fontNameMap: Map<number, Map<string, string>>;
+    private fontColorMap: Map<number, string>;
+
+    constructor(fontFamily: string = "\"Lucida Console\", Monaco, monospace") {
+        this.fontFamily = fontFamily;
+        this.fontNameMap = new Map<number, Map<string, string>>();
+        this.fontColorMap = new Map<number, string>();
+
+        PIXI.BitmapFont.from("LabelFont8Black", { fontFamily: this.fontFamily, fontSize: 8, fill: "black", fontWeight: 900 }, { chars: PIXI.BitmapFont.ASCII });
```

Comment:
This is a question, should a function "Name"-> regex like \D+(\d+)(\D+) -> {fontFamily : this.fontFamily, fontSize: $1, fill: $2, fontWeight: 900} be interesting? I see the function being called 4 times, maybe this can avoid copy-paste mistakes?

Reply:
you're right. i'll move font creation to a function e.g. createBitmapFont ( \D(\d+)(\D+) ) {...}

Reply:
I realized, we should just have the left hand side and generate the label from that. It would be so much simpler.

Reply:
i didn't quite get that...

Reply:
We have all the data, why enter it in a concatenated string to split it up again instead of giving the raw data and generating the string out of that.

Reply:
ok so just to be sure, you're referring to the label and not the font right?

Reply:
I am suggesting a "getFontName" function that takes as arguments the style
so it would be ("LabelFont" + style.fontSize + style.fill)
extra points if fill is capitalized, but as long as it's consistent, I'm happy.